### PR TITLE
Langpack Vsix packaging task and langpack refresh command.

### PR DESCRIPTION
**Pull request by smartguest**
This PR adds in a yaml task similar to the one for external extensions by packaging the current lang pack files into a drop, (Will need to be reviewed to see if this works as intended). 

Additionally, it also adds a gulp task that refreshes the lang packs with resources from XLF. Currently this task has an associated YAML compile script that is supposed to create a new pull request to merge in the changes, but this needs to be checked and tested for as well. (may need to remove the YAML script for now). 

Finally, it moves the XLF files into a folder structure that separates extension XLF files from core XLF files, (this is required for the refresh gulp task to work correctly)

### CODE REVIEW

File: build/lib/locFunc.ts [hunk 1]
```diff
@@ -0,0 +1,203 @@
+/*---------------------------------------------------------------------------------------------
```
Comment (Charles-Gagnon) [inline build/lib/locFunc.ts:1]: Was this taken from somewhere?

File: build/npm/refresh-langpack-extension.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+const textFields = {
+	"nameText": 'ads',
+	"displayNameText": 'Azure Data Studio',
+	"publisherText": 'Microsoft',
+	"licenseText": 'SEE SOURCE EULA LICENSE IN LICENSE.txt',
+	"updateText": 'cd ../vscode && npm run update-localization-extension '
+}
+
+//ADS language pack folder length
```
Comment (Charles-Gagnon) [inline build/npm/refresh-langpack-extension.js:25]: Where is this length from? The comment should explain why this value is what it is - the name already tells us that it's the "language pack folder length" so isn't really providing anything useful.

File: build/npm/refresh-langpack-extension.js [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+	if (/^\w{2}(-\w+)?$/.test(idOrPath)) {
+		locExtFolder = path.join('.', 'i18n', `ads-language-pack-${idOrPath}`);
+	}
+	let locExtStat = fs.statSync(locExtFolder);
+	if (!locExtStat || !locExtStat.isDirectory) {
+		throw new Error('No directory found at ' + idOrPath);
+	}
+	let packageJSON = JSON.parse(fs.readFileSync(path.join(locExtFolder, 'package.json')).toString());
+	//processing extension fields, version and folder name must be changed manually.
+	packageJSON['name'] = packageJSON['name'].replace('vscode', textFields.nameText);
```
Comment (Charles-Gagnon) [inline build/npm/refresh-langpack-extension.js:61]: Why are we doing this here? This seems like it should just be fixed in the package.json instead (same with the other values)

File: build/azure-pipelines/langpack-compile.yml [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    password $(github-distro-mixin-password)
+    EOF
+
+    git config user.email "sqltools@service.microsoft.com"
+    git config user.name "AzureDataStudio"
+  displayName: Prepare tooling
+
+- script: |
+    set -e
+    git checkout langpack-update
```
Comment (Charles-Gagnon) [inline build/azure-pipelines/langpack-compile.yml:30]: Why is this needed? Is this intended to actually be pushed up to github?

File: build/lib/locFunc.js [hunk 5]
```diff
@@ -0,0 +1,162 @@
+"use strict";
```
Comment (Charles-Gagnon) [inline build/lib/locFunc.js:1]: Please add xml doc comments to these functions explaining what they're doing.

**Review by Charles-Gagnon — CHANGES_REQUESTED** [review]
And FYI - if you're still working on verifying your changes (such as the yaml stuff) then I would suggest : 

1. Opening this as a draft so people know it isn't ready to be merged in yet
2. Testing stuff beforehand by queuing your own builds against the adhoc pipelines (or making your own test pipelines if needed)

File: build/npm/refresh-langpack-extension.js (hunk 2 above)
Comment (smartguest) [reply build/npm/refresh-langpack-extension.js:25]: This length is to refer to the folder section reading "ads-language-pack-", its used for isolating the ID at the end of the folder.

File: build/lib/locFunc.ts (hunk 1 above)
Comment (smartguest) [reply build/lib/locFunc.ts:1]: This is my own file that I created, using a combination of modified functions from i18n and extension.ts

File: build/azure-pipelines/langpack-compile.yml [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+  condition: and(succeeded(), eq(variables['VSCODE_STEP_ON_IT'], 'false'))
+
+- script: |
+    set -e
+    yarn gulp refresh-langpack-extension
+  displayName: Refresh Langpacks.
+
+- bash: |
+    set -e
+    # Push branch
```
[the remark this answers is not in the document]
Comment (smartguest) [reply build/azure-pipelines/langpack-compile.yml:94]: Its something that a bot should do, but I have no idea how that is supposed to be done for now.

File: build/lib/locFunc.js (hunk 5 above)
Comment (smartguest) [reply build/lib/locFunc.js:1]: Done so

**Comment by smartguest** [comment]
Will need to close this PR and reopen it with only the vsix-packaging task, refreshing the langpack requires more research into separating built in extensions and external extensions as they have significantly different ways of localizing.
